dev/PushNotifications/PushNotificationsLongRunningTask/platform.cpp

```diff
@@ -55,3 +55,25 @@ STDMETHODIMP_(HRESULT __stdcall) NotificationsLongRunningPlatformImpl::RegisterF
     return E_NOTIMPL;
 }
 
+STDMETHODIMP_(HRESULT __stdcall) NotificationsLongRunningPlatformImpl::RegisterForegroundActivator(_In_ IWpnForegroundSink* sink, _In_ PCWSTR processName)
+{
+    RETURN_HR_IF(WPN_E_PLATFORM_UNAVAILABLE, m_shutdown);
```

Comment:
Unfortunately m_shutdown is not being protected by the lock. So you can run into race conditions. @danielayala94 What is the recommended pattern here?